src/main/java/com/jvm_bloggers/entities/newsletter_issue/NewsletterIssueRepository.java

```diff
@@ -18,4 +20,8 @@
 
     List<NewsletterIssue> findAllByOrderByPublishedDateDesc();
 
+    @Query("select case when count(*) > 0 then true else false end "
+        + "from NewsletterIssue where issueNumber = :issueNumber")
+    boolean issueNumberExist(@Param("issueNumber") Long issueNumber);
```

Comment:
Very good argument against using existing method. Turns out that `NewsletterIssueRepository` extends `CrudRepository` which provides `exists(id)`method [CrudRepository::exists](http://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html?is-external=true#exists-ID-)